### ISSUE TITLE
Signing your Assemblies with a Strongname

### ISSUE THREAD

**Issue by teebeast** [QuestPDF/QuestPDF#95]
It would be fine, if you would sign your assemblies with a strong name. All our assemblies are signed with a strong name and we cannot use your assemblies without signing it with our own certificate. 

It would be a little breaking change because you have to recompile your solution. Maybe something for a new major version.

**Comment by MarcinZiabek** [comment]
You are right, signing QuestPDF assemblies would be a great idea. However, as far as I know, proper certificates are paid and pricy (I've seen offers for 250-500$ / year). I am not sure if there are any free options, do you? 

I know that in the .NET Foundation, they provide free certificates for open-source projects. However, QuestPDF was declined as not popular enough. This topic about the usefulness (in recent years) of the .NET Foundation is a different story...

**Comment by teebeast** [comment]
Look at the official documentation from [Microsoft](https://docs.microsoft.com/en-us/dotnet/standard/assembly/strong-named). It does not mean, that you sign a file. This would be nice, but not necessary and brings no breaking changes.

You can create a Key/Certificate for strong name by yourself. When you reference an assembly you can see a public key token. In .Net Core strong naming looses on meaning, especially in open source projects. Historically, it remains an issue. See this [Link](https://github.com/dotnet/runtime/blob/main/docs/project/strong-name-signing.md). So it's up to you to support it or not. For me, I can solve this problem by signing the package with my own key. I sign the assemblies in the package and put it on our private nuget feed.

By the way SkiaSharp has a strong name. ;-)

**Comment by borisdj** [comment]
Certificate for signing packages can be self-generated, they do not need to be from paid CA.
I have done it for my Bulk library, so you could do the same, more info:
[Signing the nuget package](https://github.com/borisdj/EFCore.BulkExtensions/issues/161)

**Comment by MarcinZiabek** [comment]
QuestPDF assemblies are not strong-name signed since the 2024.10.3 release. I highly appreciate your feedback and help ❤  I am closing this issue
